### PR TITLE
Edit Mode

### PR DESCRIPTION
**Pull request by vicr123**
Implement UI and logic for edit mode

### CODE REVIEW

File: BnbnavNetClient/Views/MapView.axaml.cs [hunk 1]
```diff
@@ -90,25 +104,77 @@ protected override void OnInitialized()
...
+
+        _drawnNodes = mapService.Nodes.Values.Select(node =>
+        {
+            var pos = ToScreen(new(node.X, node.Z));
+            var rect = new Rect(
+                pos.X - NodeSize / 2, 
+                pos.Y - NodeSize / 2,
+                NodeSize, NodeSize);
+            return (rect, node);
+        }).Where(node => Bounds.Intersects(node.rect)).ToList().AsReadOnly();
```
Comment (Vrabbers) [inline BnbnavNetClient/Views/MapView.axaml.cs]: Please note that there are performance implications to using LINQ and, if my memory serves me right, read-only collections too. Maybe we should reconsider and/or rewrite

Comment (vicr123) [reply BnbnavNetClient/Views/MapView.axaml.cs]: @reflectronic your thoughts?

**Comment by Vrabbers** [comment]
There's a bug.... switching to night bode and then trying to drag crashes... doesn't happen in ``main``, which is pretty werid

**Comment by Vrabbers** [comment]
Seems to be fixed but landmarks still disappear when theme switches

File: BnbnavNetClient/Views/MapView.axaml.cs (hunk 1 above)
Comment (vicr123) [reply BnbnavNetClient/Views/MapView.axaml.cs]: Performance seems to be okay so far - if required we can come back and look at this but I don't think it's an issue right now

**Pull request merged**